ynr/apps/sopn_parsing/management/commands/sopn_test_tool.py

```diff
@@ -0,0 +1,49 @@
+import requests
+import os
+
+from sopn_parsing.helpers.command_helpers import BaseSOPNParsingCommand
+
+
+class Command(BaseSOPNParsingCommand):
+    """This command uses the ballots endpoint to loop over each
+    ballot and store each sopn pdf (uploaded_file) locally"""
+
+    def handle(self, *args, **options):
+        response = requests.get(
+            "https://candidates.democracyclub.org.uk/api/next/ballots?has_sopn=1"
+        )
+        data = response.json()
+
+        pdf_files = self.get_pdfs(data)
+
+        self.save_pdfs(pdf_files)
+
+    # loop over ballots and extract uploaded_file
+    def get_pdfs(self, data):
+        pdf_files = []
+        for ballot in data["results"]:
+            if ballot["sopn"]["uploaded_file"]:
+                pdf = ballot["sopn"]["uploaded_file"]
+                path, extension = os.path.splitext(pdf)
+                if extension == ".pdf":
```

Comment:
Dont think we necessarily need to care here if the SOPN is not a PDF, as it will just mean we won't be able to parse it. But separately to this work, we might want to add some validation on the form used to upload the SOPN to check it is a PDF. That is assuming that we should only be accepting a pdf @symroe ? That's what I'm assuming, as this example of one that was uploaded as .docx and the preview isn't working https://candidates.democracyclub.org.uk/elections/local.medway.rochester-east.by.2021-12-16/sopn/

Reply:
Added a related comment to the issue as a docx file is an example of where it would like our parsing had become worse but never would have worked initially https://github.com/DemocracyClub/yournextrepresentative/issues/1377#issuecomment-976361783

Reply:
Ok, makes sense. I wonder if we should store a list of all the sopn files we can't parse, including docx?